### ISSUE TITLE
Callback is undefined on async/await function

### ISSUE THREAD

**Issue by wootsaejao** [caolan/async#1407]
Not sure this is a bug or not. It worked on my production environment right now (docker, node:7.7.0-alpine, async@2.1.5). Please guide me.

**What version of async are you using?**
2.3.0

**Which environment did the issue occur in (Node version/browser version)**
7.9.0

**What did you do? Please include a minimal reproducable case illustrating issue.**
```js
const { eachSeries } = require('async')

eachSeries( 
  [1,2,3], 
  async (item, cb) => {   // <----- it works if remove ```async```
    console.log( 'item:', item )
    console.log( 'cb:', typeof(cb) )
    console.log()
    cb()
  }, 
  err => {
    console.log('err:', err)
  }
)
```
**What did you expect to happen?**
```
item: 1
cb: function

item: 2
cb: function

item: 3
cb: function

err: null
```

**What was the actual result?**
```
item: 1
cb: undefined

err: TypeError: cb is not a function
```

**Comment by wootsaejao** [comment]
Figured out that I have to use it this way:

```js
const { eachSeries } = require('async')

eachSeries( 
  [1,2,3], 
  item => {
    console.log( 'item:', item )
    console.log()
    Promise.resolve() // <-- instead of callback
  }, 
  err => {
    console.log('err:', err)
  }
)
```

Result:
```
item: 1

item: 2

item: 3

err: null
```

**Comment by aearly** [comment]
You are correct -- the `cb` is no longer passed to `async` functions.  (Changed in 2.3.0)

That final `Promise.resolve()` is not needed.  For `async` functions, the return value becomes the result.

**Comment by wootsaejao** [comment]
Found it https://github.com/caolan/async/pull/1390/commits/faf395c546747c2066e673405e0aec94362811a0 .

Thanks.

**Comment by ktravelet** [comment]
Thank you for figuring this out :)

**Comment by sampathkolanu** [comment]
I didn't mean to hijack this thread but since i am facing a similar issue i am posting it here.

I am trying to wait 3-4 seconds after each iteratee function execution with `async.each`, since I cannot use `callback` in `async/await` function i am returning `true`

@aearly  Using `async.each` I want to delay(using `setTimeout`) the execution of code before returning in `iteratee` function **but** even before my `setTimeout` promise resolves next `iteratee` function is called with next element in the array

**Comment by aearly** [comment]
@sampathkolanu In that canse, just use something like [`delay`](https://www.npmjs.com/package/delay) and `await` it before you return, or use an inline `await new Promise(resolve => setTimeout(resolve, ms))`

**Comment by tukusejssirs** [comment]
Sorry to necrobump this thread, but could this be noted in the [docs](https://caolan.github.io/async/v3/docs.html#each)? Specifically: that if using an async iteratee, one cannot / should not use `callback()` in it.

I have encountered this issue, because `npm i async` installs version `0.9.2` and I followed the docs literally and my code worked as expected. However, then I ran `npm-check`, it said a major update is available, namely version `3.2.0`, so I upgraded it. Then the code wouldn’t work until I removed all instances of `callback()`.

Thanks for this great piece of software anyway! :+1: :wink:
